searx/settings_loader.py

```diff
@@ -1,68 +1,97 @@
+            if cfg is None:
+                cfg = {}
+            return cfg
     except IOError as e:
-        raise SearxSettingsException(e, file_name) from e
+        raise SearxSettingsException(e, str(file_name)) from e
     except yaml.YAMLError as e:
-        raise SearxSettingsException(e, file_name) from e
+        raise SearxSettingsException(e, str(file_name)) from e
+
+
+def get_yaml_cfg(file_name: str | Path) -> dict:
+    """Shortcut to load a YAML config from a file, located in the
+
+    - :py:obj:`get_user_cfg_folder` or
+    - in the ``searx`` folder of the SearXNG instalation
+    """
 
+    folder = get_user_cfg_folder()
+    if not folder:
+        folder = Path(searx_dir)
+    fname = folder / file_name
+    if not fname.is_file():
+        raise FileNotFoundError(f"File {fname} does not exist!")
 
-def get_yaml_file(file_name):
-    path = existing_filename_or_none(join(searx_dir, file_name))
-    if path is None:
-        raise FileNotFoundError(f"File {file_name} does not exist!")
+    return load_yaml(fname)
 
-    return load_yaml(path)
 
+def get_user_cfg_folder() -> Path | None:
+    """Returns folder where the user settings are located.
 
-def get_default_settings_path():
-    return existing_filename_or_none(join(searx_dir, 'settings.yml'))
+    1. If environment ``SEARXNG_SETTINGS_PATH`` is set
+       (e.g. ``/etc/mysxng/myinstance.yml``) then the path name for this
+       configuration file is the folder in which the user settings are expected
+       (e.g. ``/etc/mysxng``).  If this folder does not exists a
+       :py:obj:`EnvironmentError` is raised.
 
+    2. If folder ``/etc/searxng`` exists it is used.
 
-def get_user_settings_path() -> Optional[str]:
-    """Get an user settings file.
-    By descending priority:
-    1. ``environ['SEARXNG_SETTINGS_PATH']``
-    2. ``/etc/searxng/settings.yml`` except if ``SEARXNG_DISABLE_ETC_SETTINGS`` is ``true`` or ``1``
-    3. ``None``
+    In case none of the above path exists, ``None`` is returned.
     """
 
-    # check the environment variable SEARXNG_SETTINGS_PATH
-    # if the environment variable is defined, this is the last check
-    if 'SEARXNG_SETTINGS_PATH' in environ:
-        return existing_filename_or_none(environ['SEARXNG_SETTINGS_PATH'])
+    folder = Path("/etc/searxng")
+    settings_path = os.environ.get("SEARXNG_SETTINGS_PATH")
+    if settings_path:
+        folder = Path(settings_path).parent
+        if not folder.is_dir():
+            raise EnvironmentError(1, f"Folder of {settings_path} not exists!", folder)
 
-    # if SEARXNG_DISABLE_ETC_SETTINGS don't look any further
-    if environ.get('SEARXNG_DISABLE_ETC_SETTINGS', '').lower() in ('1', 'true'):
-        return None
+    elif os.environ.get('SEARXNG_DISABLE_ETC_SETTINGS', '').lower() in ('1', 'true'):
+        # This environment is intended exclusively for internal testing purposes
+        # and is therefore not documented!
+        folder = None
```

Comment:
```suggestion
        return None
```

Reply:
If we return here, we should also return two lines below .. None of the solutions is wrong or right, they all do what is intended. However, when implementing this function, I decided to return the value at the very end and not to exit the function within the conditions.

I'll leave it as it is for now ... or have I overlooked something?

Reply:
No, that's fine, it's just a matter of personal preference at this point, I understand your decision :)